File: app/src/main/java/sample/mail/ru/httpsample/StringUtils.java
package sample.mail.ru.httpsample;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Collection;
import java.util.Iterator;

public class StringUtils {
    public static String join(Collection<? extends CharSequence> s, String delimiter) {
        int capacity = 0;
        int delimLength = delimiter.length();
        Iterator<? extends CharSequence> iter = s.iterator();
        if (iter.hasNext()) {
            capacity += iter.next().length() + delimLength;
        }

        StringBuilder buffer = new StringBuilder(capacity);
        iter = s.iterator();
        if (iter.hasNext()) {
            buffer.append(iter.next());
            while (iter.hasNext()) {
                buffer.append(delimiter);
                buffer.append(iter.next());
            }
        }
        return buffer.toString();
    }

    public static String readInputStream(InputStream is) throws IOException {
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
        int read;
        byte[] data = new byte[16384];

        while ((read = is.read(data, 0, data.length)) != -1) {
            outputStream.write(data, 0, read);
        }

        outputStream.flush();
        return outputStream.toString("utf-8");
    }
}

File: app/src/main/java/sample/mail/ru/httpsample/StringUtils.kt
package sample.mail.ru.httpsample

import java.io.InputStream

fun String.join(s : Collection<CharSequence>, delimiter: String) : String {
    var capacity = 0
    val delimLength = delimiter.length
    var iter = s.iterator()
    if (iter.hasNext()) {
        capacity += iter.next().length + delimLength
    }

    val buffer = StringBuilder(capacity)
    iter = s.iterator()
    if (iter.hasNext()) {
        buffer.append(iter.next())
        while (iter.hasNext()) {
            buffer.append(delimiter)
            buffer.append(iter.next())
        }
    }
    return buffer.toString()
}

fun InputStream.asString() : String {
    return bufferedReader(Charsets.UTF_8).use { it.readText() }
}
